axelor-human-resource/src/main/java/com/axelor/apps/hr/service/expense/ExpenseCreateServiceImpl.java

```diff
@@ -75,6 +78,28 @@ public Expense createExpense(
     return expenseRepository.save(expense);
   }
 
+  @Override
+  @Transactional(rollbackOn = {Exception.class})
+  public Expense createExpense(User user) {
+    Expense expense = new Expense();
+    Employee employee = user.getEmployee();
+    Company company =
+        Optional.ofNullable(employee)
+            .map(Employee::getMainEmploymentContract)
+            .map(EmploymentContract::getPayCompany)
+            .orElse(user.getActiveCompany());
+    Currency currency = Optional.ofNullable(company).map(Company::getCurrency).orElse(null);
+    BankDetails bankDetails =
+        Optional.ofNullable(company).map(Company::getDefaultBankDetails).orElse(null);
+    Integer companyCbSelect =
+        Optional.ofNullable(employee).map(Employee::getCompanyCbSelect).orElse(null);
+
+    setExpenseInfo(company, employee, currency, bankDetails, null, companyCbSelect, expense);
+    expenseComputationService.compute(expense);
+
+    return expenseRepository.save(expense);
+  }
+
```

Comment:
You can create a new method using the other createExpense method, it will do more checks for bankDetails and Period.